drivers/temperature/adt7420/adt7420.h

```diff
@@ -42,92 +42,151 @@
+#define ADT7420_REG_T_LOW_LSB		    0x07 // Temperature LOW setpoint LSB
+#define ADT7420_REG_T_CRIT_MSB		    0x08 // Temperature CRIT setpoint MSB
+#define ADT7420_REG_T_CRIT_LSB		    0x09 // Temperature CRIT setpoint LSB
+#define ADT7420_REG_HIST			    0x0A // Temperature HYST setpoint
+#define ADT7420_REG_ID				    0x0B // ID
+#define ADT7420_REG_RESET			    0x2F // Software reset
 
 /* ADT7420_REG_STATUS definition */
-#define ADT7420_STATUS_T_LOW		(1 << 4)
-#define ADT7420_STATUS_T_HIGH		(1 << 5)
-#define ADT7420_STATUS_T_CRIT		(1 << 6)
-#define ADT7420_STATUS_RDY		(1 << 7)
+#define ADT7420_STATUS_T_LOW		    (1 << 4)
+#define ADT7420_STATUS_T_HIGH		    (1 << 5)
+#define ADT7420_STATUS_T_CRIT		    (1 << 6)
+#define ADT7420_STATUS_RDY			    (1 << 7)
 
 /* ADT7420_REG_CONFIG definition */
 #define ADT7420_CONFIG_FAULT_QUEUE(x)	(x & 0x3)
-#define ADT7420_CONFIG_CT_POL		(1 << 2)
-#define ADT7420_CONFIG_INT_POL		(1 << 3)
-#define ADT7420_CONFIG_INT_CT_MODE	(1 << 4)
-#define ADT7420_CONFIG_OP_MODE(x)	((x & 0x3) << 5)
-#define ADT7420_CONFIG_RESOLUTION	(1 << 7)
+#define ADT7420_CONFIG_CT_POL		    (1 << 2)
+#define ADT7420_CONFIG_INT_POL		    (1 << 3)
+#define ADT7420_CONFIG_INT_CT_MODE  	(1 << 4)
+#define ADT7420_CONFIG_OP_MODE(x)	    ((x & 0x3) << 5)
+#define ADT7420_CONFIG_RESOLUTION	    (1 << 7)
 
 /* ADT7420_CONFIG_FAULT_QUEUE(x) options */
-#define ADT7420_FAULT_QUEUE_1_FAULT	0
+#define ADT7420_FAULT_QUEUE_1_FAULT	    0
 #define ADT7420_FAULT_QUEUE_2_FAULTS	1
 #define ADT7420_FAULT_QUEUE_3_FAULTS	2
 #define ADT7420_FAULT_QUEUE_4_FAULTS	3
 
 /* ADT7420_CONFIG_OP_MODE(x) options */
-#define ADT7420_OP_MODE_CONT_CONV	0
-#define ADT7420_OP_MODE_ONE_SHOT	1
-#define ADT7420_OP_MODE_1_SPS		2
-#define ADT7420_OP_MODE_SHUTDOWN	3
+#define ADT7420_OP_MODE_CONT_CONV	    0
+#define ADT7420_OP_MODE_ONE_SHOT	    1
+#define ADT7420_OP_MODE_1_SPS		    2
+#define ADT7420_OP_MODE_SHUTDOWN	    3
 
 /* ADT7420 default ID */
-#define ADT7420_DEFAULT_ID		0xCB
+#define ADT7320_DEFAULT_ID		        0xC3
+#define ADT7420_DEFAULT_ID		        0xCB
+
 
 /******************************************************************************/
 /*************************** Types Declarations *******************************/
 /******************************************************************************/
 
+enum dev_interface {
+	SPI,
+	I2C,
+};
+
+struct adt7420_chip_info {
+	uint8_t			resolution;
+	enum dev_interface  communication;
+};
+
+enum adt7420_type {
+	ID_ADT7410,
+	ID_ADT7420,
+	ID_ADT7422,
+	ID_ADT7310,
+	ID_ADT7311,
+	ID_ADT7312,
+	ID_ADT7320
+};
+
 struct adt7420_dev {
 	/* I2C */
-	i2c_desc	*i2c_desc;
+	i2c_desc		*i2c_desc;
+	/* SPI */
+	spi_desc		*spi_desc;
+	/* Device Settings */
+	enum adt7420_type active_device;
 	/* Device Settings */
-	uint8_t		resolution_setting;
+	uint8_t		    resolution_setting;
+
 };
 
 struct adt7420_init_param {
 	/* I2C */
-	i2c_init_param	i2c_init;
+	i2c_init_param		i2c_init;
+	/* SPI */
+	spi_init_param		spi_init;
```

Comment:
You can do:
```
i2c_init_param		*i2c_init;
spi_init_param		*spi_init;
```
And create a filed in 
adt7420_dev
`bool is_spi;`
In adt7420_init you can check if  i2c_spi is not NULL and initialized the spi and set is_spi to true. Otherwise initialize i2c.
Then in the code you can just check 
`if (dev->is_spi)`
This way some extra code wont be needed

Reply:
this would be redundant information that is avoidable

a simpler way would be to have a simple static function like:
```
static bool is_spi(struct adt7420_dev *dev) {
    return chip_info[dev->active_device].communication == SPI;
}
```

It's short enough and doesn't introduce redundant desc attributes:
`if (is_spi(dev)) ...`

Reply:
A new function adt7420_spi_comm_check() has been created for the same. Used the same function call wherever the check has been made